manifests/agent/config.pp

```diff
@@ -0,0 +1,99 @@
+# == Define: aem::agent::config
+#
+# Configure a Replication Agent
+define aem::agent::config(
+  $ensure            = 'present',
+  $status            = 'enabled',
+  $description       = 'Replication Agent',
+  $on                = 'author',
+  $username          = 'admin',
+  $password          = 'admin',
+  $home              = undef,
+  $charset           = 'utf-8',
+  $loglevel          = 'info',
+  $retrydelay        = '6000',
+  $serializationtype = 'durbo',
+  $template          = '/libs/cq/replication/templates/agent',
+  $resourcetype      = 'cq/replication/components/agent',
+  $transportpassword = 'password',
+  $transporturi      = 'http://host:port/bin/receive?sling:authRequestLogin=1',
+  $transportuser     = 'replication-receiver',
+  $replicationuser   = 'your_replication_user',
+) {
+  validate_re($ensure, '^(present|absent)$', "${ensure} is not supported for ensure. Allowed values are 'present' and 'absent'.")
+
+  if $name {
+    validate_string($name)
+  }
+
+  validate_string($status)
+  validate_string($description)
+  validate_re($on, '^(author|publish)$', "${on} is not supported. Allowed values are 'author' and 'publish'.")
+  validate_string($username)
+  validate_string($password)
+  validate_absolute_path($home)
+  validate_string($charset)
+  validate_string($loglevel)
+  validate_string($retrydelay)
+  validate_string($serializationtype)
+  validate_absolute_path($template)

```

Comment:
This will allow Windows style paths, which isn't a value value for a template. Recommend just testing for it to start with a '/'